social_engagement/models.py

```diff
@@ -168,13 +168,37 @@ class StudentSocialEngagementScoreHistory(TimeStampedModel):
     score = models.IntegerField()
 
 
+class StudentSocialEngagementStats(TimeStampedModel):
+    """
+    Stores the social engagement stats for faster access and calculation.
+    """
+    score = models.OneToOneField(
```

Comment:
I don't know in detail how Django implements one-to-one fields, but if they require a table join, we might get better performance by just moving these metric fields into the actual score table itself. Denormalizing the data is a usual trick to improve performance because it avoids having to look into another table for the data. It usually leads to data replication but in this case that isn't going to happen.